ui/StreamDescriptionOverlayComponent/StreamDescriptionOverlay-Component.tsx

```diff
@@ -0,0 +1,125 @@
+import {
+    View,
+    Text,
+    Image,
+    StyleSheet,
+    Dimensions,
+  } from "react-native";
+  import React, { useState } from 'react';
+  import { Overlay } from 'react-native-elements';
+  import Icon from 'react-native-vector-icons/FontAwesome';
+  
+  type Props = {
+      stream: IStream;
+      open : boolean;
+      onButtonPress: ()=>void;
+      isFollowed: boolean;
+    };
+
+    const GetDescription=(props : Props) => {
+      return (
+      <View style={Overlaystyles.component}>
+        <Text style={Overlaystyles.component}>{props.stream.description}</Text>
+      </View>
+    );
+  };
+  const GetTitle=(props : Props) => {
+    return (
+    <View>
+      <Text style={Overlaystyles.title}>{props.stream.title}</Text>
+     </View>
+    );
+  };
+  const GetFollowed_by=(props: Props) => {
+    return (
+    <View style={Overlaystyles.component}>
+      <Text style={{color: "grey"}}><Text style={Overlaystyles.followers}>{props.stream.followed_by}</Text> Subscribers</Text>
+    </View>
+    );
+  };
+  const MyButton=(props : Props)=>{
+    const [following,changefollowing]=useState(props.isFollowed);
+    const [myIcon, setmyIcon] = following? useState("bell-slash") : useState("bell");
+    const [buttonColor, setbuttonColor] = following? useState("#1762e3") : useState("#ff0000");

```

Comment:
Instead of defining these states directly use this condition inside Icon.Button like this.
name={following ? "bell-slash" : "bell"}